source/funkin/editors/character/CharacterEditor.hx

```diff
@@ -296,8 +303,24 @@ class CharacterEditor extends UIState {
 				closeCurrentContextMenu();
 				openContextMenu(topMenu[2].childs);
 			}
+			if (Options.characterDragOffsets)
+			{
+				if (FlxG.mouse.justPressed)
+					prevDragOffsets = character.animOffsets.get(character.getAnimName()).clone();
+
+				if (FlxG.mouse.justReleased)
+				{
+					var curOffsets = character.animOffsets.get(character.getAnimName()).clone();
+					var difference = curOffsets - prevDragOffsets;
+					changeOffset(character.getAnimName(), difference.clone().negate(), false);
+					changeOffset(character.getAnimName(), difference);
```

Comment:
hm?

Reply:
reverts the offset and then add it this time to the undo list (shitty way imo but i couldnt think of a other way)

Reply:
Oh wait I'm stupid I could have used

undos.addToUndo(CChangeOffset(character.getAnimName(), difference));